src/Digdir.Domain.Dialogporten.Domain/Common/Mod11.cs

```diff
@@ -23,7 +24,14 @@ public static bool TryCalculateControlDigit(ReadOnlySpan<char> number, int[] wei
             sum += digits[i] * weights[i];
         }
         controlDigit = Mod11Number - (sum % Mod11Number);
-        return true;
+        controlDigit = controlDigit switch
+        {
+            Mod11Number => 0,
+            InvalidControlDigit => null,
+            _ => controlDigit
+        };
+
+        return controlDigit is not null;
```

Comment:
Denne regelen med at 10 er ugyldig control digit, gjelder det for både org og norsk personnummer?

Reply:
Ja, det gjør det.

Reply:
Mod11 kontroll er en algoritme som støtter kontrollsiffer fra 0-9, samt '-' dersom kontrollsifferet beregnes til 10. Både [organisasjonsnummer](https://www.brreg.no/om-oss/registrene-vare/om-enhetsregisteret/organisasjonsnummeret/), personnummer, og for øvrig bankkontonummer forkaster '-' (10) som kontrollsiffer. Dersom sum av vekttall-multiplikasjonen modulus 11 blir 0 blir også kontrollsifferet 0 (11-0=11 - så første casen). Begge disse sjekkene manglet. Jeg forsøkte å finne noe referanser på skatteetaten sine sider angående kalkulasjon av kontrollsiffer for personnummer, men fant det dessverre ikke. Personal note: det er for dårlig av skatteetaten. Fant noen andre, noe mer "shady" referanser:
- http://fnrinfo.no/Teknisk/KontrollsifferSjekk.aspx
- https://no.wikipedia.org/wiki/F%C3%B8dselsnummer

Nedenfor er et utdrag fra wikipedia (shady) som samsvarer med fnrinfo.no (shady?) og fungerer for personnummer jeg har testa med.
> Hvert av kontrollsifrene er en sjekksum beregnet ut fra de foregående sifrene. Hvert av sifrene foran multipliseres med hvert sitt tall, og resultatene legges sammen og deles på 11. Kontrollsifferet er da 11 minus resten av den siste divisjonen, med ett unntak: dersom resten er 0 og man ender opp med 11 − 0 er kontrollsifferet 0, og ikke 11.
> ...
> For noen tallkombinasjoner kan formlene for kontrollsifrene gi 10 som resultat for k1 og k2. Da kontrollsifrene er definert med kun ett siffer, blir slike personnummer forkastet.